styx-common/src/main/java/com/spotify/styx/util/ShardedCounter.java

```diff
@@ -126,23 +103,24 @@ private static void initialize(Storage storage, String counterId) {
      * </p>ex. If limit=5 for a given counter and NUM_SHARDS=3,
      * then the distribution of capacity between the 3 shards will be [2, 2, 1]
      */
-    private long shardCapacity(int shardIndex) {
+    public long shardCapacity(int shardIndex) {
       return limit / NUM_SHARDS + (shardIndex < limit % NUM_SHARDS ? 1 : 0);
     }
 
+    @Override
+    public Map<Integer, Long> getShards() {
+      return shards;
+    }
+
     /**
      * Returns shard index which _likely_ could be successfully updated by delta, according to our
      * cached view of the state in Datastore.
      */
-    private int pickShardWithSpareCapacity(long delta) {
-      List<Integer> candidates = new ArrayList<>();
-
-      for (int i : shards.keySet()) {
-        if (shards.containsKey(i) && Range
-            .closed(0L, shardCapacity(i)).contains(shards.get(i) + delta)) {
-          candidates.add(i);
-        }
-      }
+    public int pickShardWithSpareCapacity(long delta) {
+      List<Integer> candidates =
+          shards.keySet().stream().filter(index -> shards.containsKey(index) && Range
```

Comment:
The code is indented in a strange way now. Try to break the line from `.filter` already? Also I wouldn't mind having two filters to make it easier to read instead of using `&&``.